MapboxDirections/MBRouteLeg.swift

```diff
@@ -98,6 +129,11 @@ open class RouteLeg: NSObject, NSSecureCoding {
      */
     open let steps: [RouteStep]
     
+    
+    open let segmentAttributes: [Attribute:[Any]]?
```

Comment:
This property won't bridge to Objective-C, because NSDictionary's keys need to be objects, not enumeration values. (Option sets won't bridge either.) We'll probably want to use an Objective-C string enumeration (`NS_STRING_ENUM`) instead of the Swift enumeration. It would be separate from the option set used in RouteOptions.

If that's too much trouble, RouteLeg could have a separate property for each attribute. For example, `segmentSpeeds` would be of type `[CLLocationSpeed]`. That would be pretty clean, but I don't know what the story will be for any unsupported attributes we get back from the API.